docs/headerparser_usage.md

```diff
@@ -0,0 +1,77 @@
+## Headerparser
+
+headerparser is a tool that assists in writing device system call descriptions for syzkaller.
+
+In order to make Syzkaller smarter when it comes to fuzzing a device node, you can provide it with
+information about the ioctl argument struct types it expects.
+
+However, in certain cases the number of argument struct types might be high, increasing the amount of manual
+effort that goes into writing the description files for the struct types.
+
+In order to ease the effort of writing ioctl argument type description files, headerlib does a best-effort job at generating them for you. You will still need to manually select the appropriate Syzkaller data type from the list of types [here](https://github.com/google/syzkaller/blob/master/docs/syscall_descriptions_syntax.md).
+
+## Dependencies
+Headerlib uses pycparser. Please make sure you install pycparser as described below. If not you might run into a [pycparser bug](https://github.com/pyca/cryptography/issues/3187).
+
+```shell
+$ pip install --no-binary pycparser
+```
+
+## Using headerparser
+```shell
+$ python headerparser.py --device th_b --filenames=./test_headers/th_b.h
+-=============== Structures -]
+
+-=============== Structure Metadata -]
+B {
+                            B1   len|fileoff|flags|intN
+                            B2   len|fileoff|flags|intN
+}
+struct_containing_union {
+                     something   len|fileoff|flags|intN
+                 a_uion.a_char   string|filename
+                  a_uion.B_ptr   ptr|buffer|array
+}
+```
+
+You can copy paste the content underneath the `Structure Metadata` over to your syzkaller device description.
+
+## Something breaks
+Let us try parsing `test_headers/th_a.h` header file to generate argument structs.
+
+```shell
+$ python headerparser.py --device th_a --filenames=./test_headers/th_a.h
+Traceback (most recent call last):
+  File "headerparser.py", line 78, in <module>
+    main()
+	...
+	<snip>
+	...
+    raise ParseError("%s: %s" % (coord, msg))
+pycparser.plyparser.ParseError: /tmp/tmpDLK2AC/source.o:36:2: before: bool
+```
+
+From the error message, we can see that the error occurs as pycparser is not aware of the type `bool`. We can resolve this by making pycparser aware of the unknown type. In order to do this, we supply headerparser with a template file that contains C declarations and includes that can fix the parse error.
+
+```shell
+$ cat > template
+#include <stdbool.h>
```

Comment:
Should be pre-include stdbool.h automatically (or just predefine bool) if it's a common case? (and then use some other type in the example)